scout/server/extensions/loqus_extension.py

```diff
@@ -42,93 +42,213 @@ def execute_command(cmd):
-            self.loqusdb_settings,
-        )
+    def __init__(self):
+        self.loqusdb_settings = {}
+        self.loqus_ids = []
 
-    def init_app(self, app):
-        """Initialize from Flask."""
-        LOG.info("Init loqusdb app")
-        self.loqusdb_settings = self.app_config(app)
-        self.version = self.get_configured_version()
-        if not self.version:
-            self.version = self.get_version()
-        self.version_check()
-        LOG.info("Use loqusdb config file %s", self.loqusdb_settings)
+    def settings_list_to_dict(self, cfg):
+        """Convert LoqusDB list of settings into dictionary settings.
 
-    @staticmethod
-    def app_config(app):
-        """Read config.py to handle single or multiple loqusdb configurations.
+        Args:
+            cgf(list): list of dictionaries probably containing containing key/values "binary_path" and "config_path"
+        """
+        for setting in cfg:
+            cfg_id = setting.get("id") or "default"
+            self.loqusdb_settings[cfg_id] = setting
 
-        Returns: loqus_db_settings(list)"""
+    def init_app(self, app):
+        """Initialize LoqusDB from Flask app and check that settings are valid."""
+        LOG.info("Init and check loqusdb connection settings")
         cfg = app.config["LOQUSDB_SETTINGS"]
-        if isinstance(cfg, list):
-            return cfg
-        # backwards compatible, add default id
-        cfg["id"] = "default"
-        return [cfg]
 
-    def version_check(self):
+        if isinstance(cfg, list):
+            self.settings_list_to_dict(cfg)
+            LOG.warning(
+                "Deprecated settings: Scout version >=5 will no longer accept LoqusDB settings defined as a list. For additional info please check the Scout admin guide."
+            )
+        elif isinstance(cfg, dict) and "default" not in cfg:
+            self.loqusdb_settings["default"] = cfg
+            LOG.warning(
+                "Deprecated settings: Scout version >=5 will no longeg accept LoqusDB settings missing the instance ID. For additional info please check the Scout admin guide."
+            )
+        else:
+            self.loqusdb_settings = cfg
+
+        for key, setting in self.loqusdb_settings.items():
+            # Scout might connect to Loqus via an API or an executable, define which one for every instance
+            setting["instance_type"] = "api" if setting.get(API_URL) else "exec"
+            if app.config["TESTING"] is True:
+                setting["version"] = 2.5
+            else:
+                setting["version"] = self.get_instance_version(setting)
+            self.version_check(setting)
+
+        self.loqus_ids = self.loqusdb_settings.keys()
+        LOG.info(f"LoqusDB setup: {self.__repr__()}")
+
+    def version_check(self, loqusdb_settings):
         """Check if a compatible version is used otherwise raise an error"""
-        if not self.version >= 2.5:
```

Comment:
I don't think that assigning a default version to the loqus instance makes sense, since it can include many loqus instances, each with a different version (all >2.5)